### PR TITLE
merge lock, key, locks, account validators from paywall

### PR DESCRIPTION
**Pull request by cellog**
# Description

In preparation for the `postOfficeService`, this copy/pastes the validators needed for the list of locks that will come in to the account iframe

# Issues

Refs #3764 

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/utils/validators.js [hunk 1]
```diff
@@ -19,3 +27,130 @@ export const isPositiveNumber = val => {
   const parsedFloat = parseFloat(val)
   return !isNaN(parsedFloat) && +parsedFloat >= 0
 }
+
+export const isAccountOrNull = val => {
+  return val === null || (typeof val === 'string' && val.match(ACCOUNT_REGEXP))
```
Comment (akeem) [inline unlock-app/src/utils/validators.js]: could this call `isAccount` for the right hand side of the || ?

File: unlock-app/src/utils/validators.js [hunk 2]
```diff
@@ -19,3 +27,130 @@ export const isPositiveNumber = val => {
...
+
+  if (keys.length < validKeys.length) return false
+  if (validKeys.filter(key => !keys.includes(key)).length) return false
+  return true
+}
+
+/**
+ * validate a single key. This should match the type in unlockTypes.ts
+ */
+export const isValidKey = key => {
```
Comment (akeem) [inline unlock-app/src/utils/validators.js]: There is a lot going on in this method, lets see if we can break this down a bit, totally fine if we extract some supporting methods here

File: unlock-app/src/utils/validators.js [hunk 3]
```diff
@@ -19,3 +27,130 @@ export const isPositiveNumber = val => {
...
+  }
+  */
+  return true
+}
+
+/**
+ * validate a single lock. This should match the type in unlockTypes.ts
+ */
+export const isValidLock = lock => {
+  if (
```
Comment (akeem) [inline unlock-app/src/utils/validators.js]: Same as the method above

**Comment by cellog** [comment]
PTAL @akeem

**Pull request merged**